Entrega da Segunda Iteração : TA

psg2 opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pauloborba/ta/19)

<!-- Reviewable:end -->

dansq committed:
Test implementation for RegisterDiscipline.

Changes in .feature file following monitor's observations (removed ":" from Given, When, Then and And).

Created auxiliary .groovy files for the test.

dansq committed:
Merge remote-tracking branch 'origin/master'

Conflicts:
	test/cucumber/RegisterDiscipline.feature

Commit:
Passos de teste
Classes auxiliares
Adaptação dos cenários de feature

Commit:
Ajustes finais

Commit:
Remoção de ações em branco.

Commit:
Created Sheet and SheetBuilder and made small corrections to ImportConcepts and ConceptController.

renatodms committed:
Groovy de testes da Feature RegistrationOfTeachers

imvm committed:
Merge pull request #3 from renatodeyvson/master

Groovy de testes da Feature RegistrationOfTeachers

imvm committed:
Beggining notification feature test implementation

imvm committed:
Beggining notification feature test implementation

imvm committed:
Merge branch 'master' of https://github.com/imvm/TA

cjvg committed:
Merge pull request #3 from gaabs/master

Import Concepts from SpreedSheet Test + auxiliary classes

cjvg committed:
Changes for auto evaluation

cjvg committed:
Change for auto evaluation 2

cjvg committed:
Merge remote-tracking branch 'origin/master'

Commit:
Update EvaluationComparison.groovy

Commit:
Update EvaluationComparison.groovy

Commit:
Update ShowComparisonPage.groovy

Commit:
Update EvaluationComparison.groovy

Commit:
Update EvaluationComparison.groovy

Commit:
Update ManualConceptInput.feature

Commit:
Update ManualConceptInput.feature

Commit:
Update ManualConceptInput.feature

Commit:
Update ManualConceptInputSteps.groovy

Commit:
Update ManualInputPage.groovy

Commit:
Edited chromedriver file and gebconfig to make tests run.

Corrected a few errors on features, tests and auxiliary classes from ImportConcepts feature.

diest committed:
Correções

Deixei minha feature melhor apropriada para a implementação de seu teste.

Commit:
Update EvaluationComparison.groovy

dansq committed:
Merge pull request #4 from ehammo/master

Merge de testes

cjvg committed:
Merge pull request #4 from gaabs/master

Edited chromedriver file and gebconfig to make tests run.

imvm committed:
Implementacao testes de notificacao

diest committed:
Create GradeCalculator.groovy

Esta classe só tem o método "vazio" calculateGrade para não gerar erro nos testes.

diest committed:
Create FinalGradeController.groovy

Uma interface para o método do qual precisarei nos testes. Ainda preciso incrementar mais para os testes não comprometerem nenhuma funcionalidade.

diest committed:
Create FinalGradeSteps.groovy

Primeira parte da criação dos testes da feature

diest committed:
Update FinalGrade.feature

Mais edições para melhorar a parte dos testes

diest committed:
Mais correções

Corrigi mais imprecisões dos cenários

diest committed:
Update FinalGrade.feature

dansq committed:
Correção de typos/nomes de classes

diest committed:
Delete GradeCalculator.groovy

Não era necessário

dansq committed:
Merge remote-tracking branch 'origin/master'

diest committed:
Interfaces incrementadas

Adicionadas mais interfaces para que o funcionamento não seja prejudicado.

diest committed:
Update FinalGradeSteps.groovy

Não fiz testes de GUI porque não entendi como minha feature deve se comportar nas páginas ou as ações que ela deva praticar em cada passo.

diest committed:
Correção de FinalGradeSteps.groovy

Corrigi alguns erros básicos

renatodms committed:
Merge pull request #1 from dansq/master

Testes groovy

renatodms committed:
Merge branch 'imvm-master'

renatodms committed:
Merge https://github.com/renatodeyvson/TA

Commit:
Merge pull request #6 from cjvg/master

Rearranjo de envio

Commit:
Merge pull request #2 from renatodeyvson/master

3o fase

Commit:
Merge branch 'master' of https://github.com/psg2/TA into psg2-master

Commit:
Update GebConfig.groovy

Commit:
Update GebConfig.groovy

Commit:
Update GebConfig.groovy

Commit:
Merge pull request #3 from ehammo/master

Merged

Commit:
Merge pull request #4 from diest/master

Commits do aluno 10

Commit:
Corrected many compilation errors.

Tried everything, but couldn't make Tests back to work since merge.

Commit:
Commented whole class AlternativeControllerSpec. It had incompatible commands to our grails version.

Full commented 	accesscontrol and              	autoavaliacaoprazo.

corrected pedro duplicate steps

corrected renato duplicate steps

corrected caio duplicate steps

corrected maia duplicate steps and classes

corrected ian notifications feature and autoavaliacaoprazo

deleted	cadastrodecriterios feature (unknown format)

dansq committed:
Merge remote-tracking branch 'upstream/master'

imvm committed:
Implementação feature Notifications

dansq committed:
entrega features, correção controlador, views, e domain

Commit:
Merge pull request #7 from gaabs/master

Pegar fork mais atualizado

Commit:
Domain class "Concept" created.

Commit:
Concept views

dansq committed:
arquivo foi modificado na máquina virtual, voltando ao original

jgsjv committed:
Merge pull request #6 from dansq/master

Merge da entrega 21/10/2015

Commit:
Atualizando repositório

imvm committed:
Delete modules.xml

imvm committed:
Setting Travis build

Commit:
Feature de inserção de conceitos manuais nos critérios.

imvm committed:
Deleting configuration file

imvm committed:
Merge https://github.com/imvm/TA

Commit:
Implemented scenario: importing invalid spreadsheet (gui and controller)

Commit:
Merge pull request #6 from psg2/master

Merge 3 >> 4

Commit:
Domain

Commit:
Backup

dansq committed:
Merge pull request #7 from jgsjv/master

Merge pull request #6 from dansq/master

Commit:
merge

dansq committed:
Merge branch 'igormatos-master'

Commit:
commit module

Commit:
Ian Merge

Commit:
Merge branch 'imvm-master'

Commit:
Merge Dansq

Commit:
Merge branch 'dansq-master'

Commit:
Merge branch 'master' of https://github.com/dansq/TA into dansq-master

Conflicts:
	.idea/modules.xml
	grails-app/controllers/ta/ConceptController.groovy
	grails-app/controllers/ta/NotificationController.groovy
	grails-app/controllers/ta/StudentController.groovy
	grails-app/views/student/index.gsp
	test/cucumber/AccessControl.feature
	test/cucumber/AutoavaliacaoPrazo.feature
	test/cucumber/steps/FinalGradeSteps.groovy
	test/functional/pages/ShowNotificationsPage.groovy
	test/functional/steps/NotificationsTestDataAndOperations.groovy

Commit:
Merge branch 'dansq-master'

Commit:
Correção de Conflitos

Commit:
Merge gaabs

Commit:
Merge branch 'gaabs-master'

Commit:
Merge branch 'master' of https://github.com/gaabs/TA into gaabs-master

Conflicts:
	grails-app/controllers/ta/ConceptController.groovy
	grails-app/views/concept/index.gsp
	grails-app/views/concept/upload.gsp

Commit:
Merge branch 'gaabs-master'

Commit:
Correção de Conflitos numero 2

Commit:
Comentários em testes com duplicate steps.

imvm committed:
geb config file

imvm committed:
geb config

imvm committed:
geb config

imvm committed:
pull from fork

imvm committed:
test adjustments

Commit:
Ajustes no Travis, testes e controller'

Commit:
Correção import errado

Commit:
Importação de planilha ok. Falta apenas adicionar os dados importados no estudante, dado que eles está mal estruturado.

Commit:
Correção de testes

Commit:
Alteraçao no caso de arquivo 'csv', pois o plugin nao importa arquivos nesse formato.

Commit:
Concept update, modifications in domain Student and controller Student.
Small updates in Student views as well

Commit:
alteracoes no formato planilha

Commit:
Calculo de media dos criterios

Commit:
Merge branch 'master' of https://github.com/gaabs/TA

Commit:
Merge branch 'Testes' of https://github.com/psg2/TA

Commit:
Alteração na importação de planilhas.
Caso o aluno ou criterio da planilha não exista, ele é cadastrado.

Commit:
Implementacao de Testes

cjvg committed:
Merge pull request #6 from psg2/Testes

Testes

Commit:
Alterações na feature de importação de planilhas
Alteração na importação de planilhas.
Verificação de colunas válidas

Commit:
Adição de form para view e metodo para o submit

Commit:
Adição de cenários e implementação de testes

Commit:
Adição de métodos auxiliares para teste de controller e GUI

Commit:
Correção do caso de erro da GUI

cjvg committed:
Mudanças para feature, steps e student

Commit:
Alterações na feature de importação de planilhas
Alteração na importação de planilhas (conceptController)
Verificação de colunas válidas (sheetImporter)
Alteração nos testes de importação de planilhas

Commit:
Adicionei arquivos auxiliares para testes (planilhas padrão e defeituosas)
Correção e implementação de testes da importação de planilhas

Commit:
Merge branch 'Testes' of https://github.com/psg2/TA

Conflicts:
	test/cucumber/steps/ManualConceptInputSteps.groovy

cjvg committed:
shows auto evaluation

cjvg committed:
changes to update auto evaluation

Commit:
Alterações na importação de planilhas, correção e implementação de respectivos testes

Commit:
Leitura JQuery da página de inserção de conceitos para testes de GUI
Alguns ajutes para o Controller e views de Student

Commit:
Compare Feature

Commit:
Configuração da IDE para Windows
Mudança dos títulos das páginas
Teste GUI semi-completo

Commit:
Adicionei arquivos auxiliares para testes (planilhas padrão e defeituosas)
Correção e implementação de testes da importação de planilhas

Commit:
gebconfig windows

Commit:
Merge remote-tracking branch 'origin/master'

Conflicts:
	test/functional/GebConfig.groovy

Commit:
Compare Feature steps

Commit:
Steps changes

Commit:
ImportConcepts from spreadsheet GUI tests

Commit:
files from gui test (sheet import)

Commit:
Steps and .feature changes
Pages and views updates

Commit:
ImportConcepts from spreadsheet  tests

Commit:
coloring differences

Commit:
-unit

Commit:
Finalização dos 8 cenários de teste.

cjvg committed:
changed methods for auto evaluation and the page

Commit:
Correção e implementação de testes da importação de planilhas

Commit:
[completed #107523732] [completed #109086474] Finalização dos 8 testes da Feature Manual Concept Input

Commit:
Tentativa de corrigir falhas nos testes. Há um problema de alteração concorrente no DB ainda não resolvido, e, aparentemente, o Travis não consegue compilar o Excel Importer.

Commit:
travis verbose

Commit:
test updates

imvm committed:
updating branch

Commit:
Adição de css para página da feature

Commit:
Steps Comentado

Commit:
Merge pull request #7 from ehammo/Testes

Steps Comentado

Commit:
Merge pull request #14 from psg2/Testes

Merge com branch

Commit:
Para o travis

Commit:
Para o travis - commit 2

cjvg committed:
[109091068] controller test for auto evaluation

dansq committed:
adicionando novos cenários e testes para RegisterDiscipline

renatodms committed:
atualização de dados do professor

renatodms committed:
Cenários de atualização de dados de professores

renatodms committed:
Merge pull request #3 from psg2/master

atualizar

imvm committed:
travis fix

imvm committed:
merge renato

imvm committed:
removed duplicated steps

imvm committed:
geb fix

dansq committed:
merge

dansq committed:
merge

cjvg committed:
Merge branch 'master' of https://github.com/gaabs/TA into gaabs-master

# Conflicts:
#	grails-app/controllers/ta/StudentController.groovy

cjvg committed:
Merge branch 'gaabs-master'

dansq committed:
erge

dansq committed:
Merge branch 'ehammo-master'

imvm committed:
Adjusting Notification controller and domain classes

imvm committed:
Fixed incorrect name error

imvm committed:
Removed duplicated steps

imvm committed:
Organized steps file

imvm committed:
Removed duplicate step, organized steps file

imvm committed:
Removed unnecessary link

imvm committed:
[#107883384]Developing scenario test

imvm committed:
[#107883384]Updated feature file

imvm committed:
[#107883384]Updated Notification class attributes

imvm committed:
[#107883384]Updated feature file

imvm committed:
[#107883384]Developing scenario tests

imvm committed:
[#107883412]Developing scenario tests

imvm committed:
[#107883384]Developing scenario tests

imvm committed:
[#107883384][#107883412]Updating steps

imvm committed:
[#107883384][#107883412]Changed test student name and login

imvm committed:
[#107883384][#107883412]Changed test parameters

imvm committed:
[#107883384][#107883412]Changed test parameters

imvm committed:
[#107883384][#107883412]Changed test parameters

imvm committed:
Added ignore

imvm committed:
[#107883384][#107883412]Developing tests

imvm committed:
Reverted local testing change

imvm committed:
Fixed ChromeDriver import

imvm committed:
Fixing Travis invalid variable bug

imvm committed:
Fixing comment

imvm committed:
Removed comment

Commit:
merge cjvg

Commit:
Merge branch 'cjvg-master'

Commit:
[#107523050] + Merge imvm

Commit:
Merge branch 'imvm-master'

Commit:
Update ImportConcepts.feature

Ignorando cenários por conta da incompatibilidade do Travis com o plug-in do Excel utilizado

Commit:
para o travis - 1 merge

Commit:
Merge pull request #8 from dansq/master

2a fase da entrega

Commit:
para o travis - 2 merge

Commit:
para o travis - 2 merge - corrigindo erros de compilacao do RegisterDiscipline.groovy

Commit:
para o travis - 2 merge - corrigindo erros de compilacao do RegisterDiscipline.groovy

renatodms committed:
Merge pull request #4 from gaabs/master

Entrega 2a iteração: 1 + 5 >> 7

Commit:
@ignore nos meus testes e modificações para compatibilizar com testes de contribuidores

Commit:
Merge branch 'master' of https://github.com/gaabs/TA

Conflicts:
	test/cucumber/ImportConcepts.feature

Commit:
Tentativa de compatibilizar testes da equipe

Commit:
Tentativa de compatibilizar testes da equipe

Commit:
Resolução de conflitos

Commit:
Resolução para merge

Commit:
Merge branch 'ehammo-master'

Commit:
Correção de conflitos remanescentes do merge.
Correção do GebConfig para rodar no Travis
Adição do método deleteAfterTest()

renatodms committed:
Merge pull request #5 from gaabs/master

Pull request com alterações

Commit:
Correção do método deleteAfterTest

Commit:
Correção de conflitos para segundo merge

Commit:
Merge branch 'renatodeyvson-master'

Commit:
Correção de conflitos pós merge.

pauloborba commented on .idea/vcs.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
tirar esses arquivos do commit

pauloborba commented on test/cucumber/ManualConceptInput.feature:
```diff
@@ -0,0 +1,99 @@
+    And the student doesn't have a concept in that criterion
+    When the user input manually a new concept "MA" into the student in that criterion
+    Then the new concept of that criterion is stored in the student
+    And the final criterion concept of that student is updated to "MA" in the system
```
fora as dependencias, esta ok

pauloborba commented on test/cucumber/ManualConceptInput.feature:
```diff
@@ -0,0 +1,99 @@
+    And the final criterion concept of that student is updated to "MA" in the system
+    
+  @psg2
+  Scenario: Student with no concepts with manual input MPA
```
evitar a duplicação

pauloborba commented on test/cucumber/ManualConceptInput.feature:
```diff
@@ -0,0 +1,99 @@
+    When I go to the Manual Input Concept Page
+    And I choose a new concept "MPA" to that student in that criterion
+    And I submit the info
+    Then I go back to Student page
```
when

pauloborba commented on test/cucumber/steps/ManualConceptInputSteps.groovy:
```diff
@@ -0,0 +1,148 @@
+}
+
+And (~'I already put the concepts "([^"]*)" and "([^"]*)" for that student in that criterion$'){String concept1, concept2 ->
+    to ManualInputPage
```
evitar duplicação

pauloborba commented on test/functional/pages/StudentPages/StudentPage.groovy:
```diff
@@ -1,12 +1,94 @@
+
+        String test = $(idCriteria).text()
+
+        if ( !test.contains(",") ){
```
?

pauloborba commented on grails-app/controllers/ta/StudentController.groovy:
```diff
@@ -102,18 +183,99 @@ class StudentController {
-        System.out.println(params.get("concept"))
+    def updateAutoEvaluation(String login, String studentCriterion, String concept) {
+        if (!concept.isEmpty()) {
+            println login
```
?

pauloborba commented on test/cucumber/ImportConcepts.feature:
```diff
@@ -0,0 +1,75 @@
+  Scenario: Importing spreadsheet with invalid column
+    Given that I am at the Sheet Upload page
+    When I import the spreadsheet "invalidColumnSheet.xlsx"
+    And the spreadsheet is on valid file format
```
given

pauloborba commented on test/cucumber/ImportConcepts.feature:
```diff
@@ -0,0 +1,75 @@
+  Scenario: Importing valid spreadsheet (file format and columns)
+    Given the spreadsheet "validSheet.xlsx" is on valid file format
+    When I try to import its data
+    And the spreadsheet contains valid columns
```
given

pauloborba commented on test/cucumber/steps/ImportConcepts.groovy:
```diff
@@ -0,0 +1,212 @@
+
+// Then update system data accordingly
+Then (~'^update system data accordingly$'){ ->
+	assert sheetController.hasImported
```
melhor evitar isso

pauloborba commented on test/cucumber/AutoEvaluationEdit.feature:
```diff
@@ -0,0 +1,25 @@
+  Scenario: edit auto evaluation
+    Given there is the student "Peter Parker" with login "pp"
+    And the criterion "Requirements"
+    And the student does not have an auto evaluation for that criterion
```
dependencia

pauloborba commented on test/cucumber/AutoEvaluationEdit.feature:
```diff
@@ -0,0 +1,25 @@
+    When I go to the Auto Evaluation Page
+    And I choose the student name
+    And I choose a new concept "MA" to that student for that criterion
+    And I send the info
```
nivel abstracao

pauloborba commented on test/cucumber/AutoEvaluationEdit.feature:
```diff
@@ -0,0 +1,25 @@
+    And I choose the student name
+    And I choose a new concept "MA" to that student for that criterion
+    And I send the info
+    Then I go to that student page
```
I am at ...

pauloborba commented on test/cucumber/Notifications.feature:
```diff
@@ -0,0 +1,43 @@
+Scenario: Registering a grade that requests a notification
+	Given that the system has a student named "Nicolla Henrique" with login "nhps" registered
+	And that the system has evaluation criteria named "Hula-hooping", "Jump Rope", and "Crosswords" registered
+	And that "nhps" only has a MANA registered as a grade for the "Hula-hooping" and "Jump Rope" criteria
```
MANA deveria ser parametro

pauloborba commented on grails-app/controllers/ta/NotificationController.groovy:
```diff
@@ -0,0 +1,32 @@
+    public boolean saveNotification(Notification notification) {
+        if (Notification.findByLogin(notification.login) == null) {
+            notification.save(flush: true)
+            new StudentController().updateStudentEvaluationCriteria()
```
?

pauloborba commented on test/cucumber/EvaluationComparison.feature:
```diff
@@ -0,0 +1,37 @@
+#Success
+  @ehammo
+Scenario: table is seen with success
+Given There is a student with the login "rp" and name "Ray Palmer" and a criteria with name "C1" and the student appear in the list of student that sent their auto-Evaluation
```
quebrar

pauloborba commented on test/cucumber/EvaluationComparison.feature:
```diff
@@ -0,0 +1,37 @@
+Scenario: table is seen with success
+Given There is a student with the login "rp" and name "Ray Palmer" and a criteria with name "C1" and the student appear in the list of student that sent their auto-Evaluation
+When I choose to compare the grades of the student with the login "rp"
+Then I can see a detailed table with both student and the professor Evaluations being put, in each criterion, side by side in the screen
```
detalhar

pauloborba commented on test/cucumber/steps/EvaluationComparisonSteps.groovy:
```diff
@@ -0,0 +1,103 @@
+}
+
+Then (~'^I can see a detailed table with both student and the professor Evaluations being put, in each criterion, side by side in the screen$'){->
+   at StudentPage
```
falta detalhar

pauloborba commented on test/cucumber/EvaluationComparison.feature:
```diff
@@ -0,0 +1,37 @@
+Scenario:  table is not seen with success
+Given The student with the login "sk" and name "Selina Kyle" and a criteria with name "C1" do not appear in the list of student that sent their auto-Evaluation
+When I choose to compare the grades of the student with the login "sk"
+Then I should stay in the Student page
```
error message

pauloborba commented on test/cucumber/EvaluationComparison.feature:
```diff
@@ -0,0 +1,37 @@
+#Failure
+  @ehammo
+Scenario:  table is not seen with success
+Given The student with the login "sk" and name "Selina Kyle" and a criteria with name "C1" do not appear in the list of student that sent their auto-Evaluation
```
given I'm at

pauloborba commented on test/cucumber/EvaluationComparison.feature:
```diff
@@ -0,0 +1,37 @@
+  @ehammo
+Scenario: The system return a table with success
+Given The Auto-Evaluation of the student with the login "dp" and name "Diana Prince" in the criteria with name "C1" is on the database
+When The system requires the Evaluation and Auto-evaluation comparison of student with the login "dp"
```
there is a request

pauloborba commented on test/cucumber/EvaluationComparison.feature:
```diff
@@ -0,0 +1,37 @@
+Scenario: The system do not return a table with success
+Given The Auto-Evaluation of the student with the login "ac" and name "Arthur Curry" in the criteria with name "C1" is not on the database
+When The system requires the Evaluation and Auto-evaluation comparison of student with the login "ac"
+Then The system returns an error message
```
nao alteram o estado

pauloborba commented on test/cucumber/steps/EvaluationComparisonSteps.groovy:
```diff
@@ -0,0 +1,103 @@
+}
+
+Then (~'^The system returns an error message$'){->
+    assert !studentX.worked
```
evitar

pauloborba commented on test/cucumber/steps/EvaluationComparisonSteps.groovy:
```diff
@@ -0,0 +1,103 @@
+}
+
+Then (~'^The system returns a detailed table with both student and the professor grades$'){->
+    assert studentX.worked
```
revisar

pauloborba commented on grails-app/controllers/ta/StudentController.groovy:
```diff
@@ -39,12 +68,66 @@ class StudentController {
+        String login = params.studentId
+        def criteria = EvaluationCriterion.findAll()
+        Student student = Student.findByLogin(login)
+        HashMap<String, String> auto = student.getAutoEvaluations()
```
nao usando

pauloborba commented on grails-app/controllers/ta/StudentController.groovy:
```diff
@@ -39,12 +68,66 @@ class StudentController {
+        render view: "compare", model: [criteria: criteria, student: student]
+    }
+
+    def compareGrades(String login) {
```
duplicacao

pauloborba commented on test/cucumber/RegisterDiscipline.feature:
```diff
@@ -0,0 +1,96 @@
+  @ignore
+  Scenario: new discipline
+
+    Given I am at the homepage
```
e não consigo ver a disciplina...

pauloborba commented on test/cucumber/RegisterDiscipline.feature:
```diff
@@ -0,0 +1,96 @@
+
+    When I select create new discipline
+    And fill the form with name "Engenharia de Software e Sistemas" with teacher "Paulo Borba" and concepts "MA, MPA, MANA"
+    And the system already has a discipline named "Engenharia de Software e Sistemas"
```
given

pauloborba commented on test/cucumber/RegisterDiscipline.feature:
```diff
@@ -0,0 +1,96 @@
+Scenario: Discipline with no teacher
+
+	Given the system has no discipline named "ESS"
+	When I create the discipline "ESS" with concepts "MA, MPA, MANA"
```
and no teacher name

pauloborba commented on test/cucumber/RegisterDiscipline.feature:
```diff
@@ -0,0 +1,96 @@
+	And fill the form with name "ESS" with concepts "MA, MPA, MANA"
+	And do not fill in the teacher
+
+	Then I'm not able to save the discipline until I fill in the teacher
```
mesanegem de erro?

pauloborba commented on test/cucumber/FinalGrade.feature:
```diff
@@ -0,0 +1,37 @@
+  Scenario: Inability to calculate grade
+    Given that at least one criteria has no grades for student "Eduardo"
+    When I request the system to register grades from "Eduardo"
+    Then the final grade is not calculated
```
muda nada

pauloborba commented on test/cucumber/FinalGrade.feature:
```diff
@@ -0,0 +1,37 @@
+    And I can see a table relating students to their grades on each criteria
+    And one of the sections is named "Media final"
+    When a student has a grade shown in each criteria
+    Then "Media final" displays the student's final grade
```
amarra valor aqui ou nos cenários de pedro?

pauloborba commented on test/cucumber/FinalGrade.feature:
```diff
@@ -0,0 +1,37 @@
+    Given that I am at student page
+    And I can see a table relating students to their grades on each criteria
+    And one of the sections is named "Media final"
+    When a student has at least one criteria without a grade
```
given

pauloborba commented on test/cucumber/FinalGrade.feature:
```diff
@@ -0,0 +1,37 @@
+    And I can see a table relating students to their grades on each criteria
+    And one of the sections is named "Media final"
+    When a student has at least one criteria without a grade
+    Then "Media final" displays the message Avaliacoes insuficientes
```
parametro?

mensagem

pauloborba commented on test/cucumber/AccessControl.feature:
```diff
@@ -0,0 +1,35 @@
+#+  I want to have access to the system
+#+  So that I can see the options that belongs to my user type
+#+
+#+  #Controller scenario
```
@ignore, ao invés de comentar

pauloborba commented on test/cucumber/AccessControl.feature:
```diff
@@ -0,0 +1,35 @@
+#+  Scenario: Signing in
+#+    Given that "joao" has a account with password "123"
+#+    When "joao" request to sign in with his correct password "123"
+#+    Then "joao" have access to the system
```
have -> has

pauloborba commented on test/cucumber/AccessControl.feature:
```diff
@@ -0,0 +1,35 @@
+#+    Given that "joao" has a account with password "123"
+#+    When "joao" request to sign in with his correct password "123"
+#+    Then "joao" have access to the system
+#+    And the session is saved on his browser cookies
```
didicil testar

renatodms committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	test/cucumber/RegistrationOfTeachers.feature

jpms2 committed:
Merge pull request #1 from gaabs/master

tentativa de passar teste do excel

jpms2 committed:
Update ImportConcepts.feature

jpms2 committed:
Update BuildConfig.groovy

Commit:
Merge pull request #9 from jpms2/master

Correção travis-planilha

Commit:
Merge https://github.com/gaabs/TA

Commit:
Teste travis

Commit:
Remoção do CSS
Remoção dos critérios removidos das hash contidas nos alunos
Criação da classe getPageTitle para os testes de GUI serem compatíveis com qualquer língua disponível
Refatoração de código duplicado em Student na inicialização das hashs.
Refatoração de código duplicado em StudentPage para verificação de conceitos e critérios nos testes de GUI.
Correção de um teste antigo com erro de digitação

Commit:
[#110668934] Refatoração da importação de conceitos (planilha)
- Correção dos testes de planilha com Travis
- Adição da limpeza do BD antes de cada teste
- Refatoração de SheetController, SheetImporter, ImportConcepts.feature, ImportConcepts.groovy.
- Adição da classe auxiliar para testes de GUI GetPageTitle
- Adição da planilha para testes ValidSheet2

renatodms committed:
Testes de RegistrationOfTeachers

Commit:
[#111367600] Adição de novos cenários de GUI
Ajuste de cenários antigos para melhorar a complexidade
Ajuste nas pages para funcionar com Locale e GetPageTitle
Correção da configuração de limpar o banco de dados após cada teste para funcionar com GUI

renatodms committed:
Merge pull request #7 from gaabs/master

Atualização

renatodms committed:
Testes de atualização de professor

renatodms committed:
Correção de testes

Commit:
fixed untracked files

Commit:
Merge https://github.com/pauloborba/TA

Conflicts:
	test/functional/GebConfig.groovy

Commit:
Merge https://github.com/pauloborba/TA

Conflicts:
	test/functional/GebConfig.groovy

Pages corrections

Commit:
[#110668934] Refatoração da SheetUploadPage

Commit:
[#110668934] Refatoração

renatodms committed:
Correção de testes (Teachers)

Commit:
[completed #110668934] Finalização da Refatoração da importação de conceitos
- Remoção de método inutilizado click() na SheetUploadPage
- Adição do diretório ".slcache" ao gitignore

Commit:
gebConfig

Commit:
Merge master

Commit:
Merge pull request #17 from pauloborba/master

Merge master

Commit:
[ Completed #111367600 #111367594 ] Adição de novos cenários, correção de antigos, refatoração e ajustes nas Pages.

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Travis Update

Commit:
Merge https://github.com/pauloborba/TA
Travis fix

# Conflicts:
#	.travis.yml

renatodms committed:
Cenários de atualização de nome de professor

Commit:
Merge branch 'master' of https://github.com/psg2/TA into psg2-master

# Conflicts:
#	.idea/modules.xml
#	test/cucumber/EvaluateStudent.feature
#	test/cucumber/ManualConceptInput.feature
#	test/cucumber/steps/ManualConceptInputSteps.groovy
#	test/cucumber/support/env.groovy
#	test/functional/GebConfig.groovy
#	test/functional/pages/EvaluationCriterionPages/CreateEvaluationCriterionPage.groovy
#	test/functional/pages/EvaluationCriterionPages/EvaluationCriterionPage.groovy
#	test/functional/pages/GetPageTitle.groovy
#	test/functional/pages/ManualInputPage.groovy
#	test/functional/pages/RegisterEvaluationPage.groovy
#	test/functional/pages/ShowEvaluationPage.groovy
#	test/functional/pages/StudentPages/CreateStudentPage.groovy
#	test/functional/pages/StudentPages/StudentPage.groovy

Commit:
Merge branch 'master' of https://github.com/psg2/TA into psg2-master

# Conflicts:
#	.idea/modules.xml
#	test/cucumber/EvaluateStudent.feature
#	test/cucumber/ManualConceptInput.feature
#	test/cucumber/steps/ManualConceptInputSteps.groovy
#	test/cucumber/support/env.groovy
#	test/functional/GebConfig.groovy
#	test/functional/pages/EvaluationCriterionPages/CreateEvaluationCriterionPage.groovy
#	test/functional/pages/EvaluationCriterionPages/EvaluationCriterionPage.groovy
#	test/functional/pages/GetPageTitle.groovy
#	test/functional/pages/ManualInputPage.groovy
#	test/functional/pages/RegisterEvaluationPage.groovy
#	test/functional/pages/ShowEvaluationPage.groovy
#	test/functional/pages/StudentPages/CreateStudentPage.groovy
#	test/functional/pages/StudentPages/StudentPage.groovy

Commit:
Merge branch 'psg2-master'

renatodms committed:
Refactoring RegistrationOfTeachers

Commit:
Merge pull request #18 from gaabs/master

Última iteração: 6 >> 5

Commit:
Merge Renatodeyvson

Commit:
Resolving conflict

Commit:
Merge branch 'renatodeyvson-master'
